Source/Evolve/WarpXComputeDt.cpp

```diff
@@ -58,16 +60,15 @@ WarpX::ComputeDt ()
 
     if (m_const_dt.has_value()) {
         deltat = m_const_dt.value();
-    } else if (electromagnetic_solver_id == ElectromagneticSolverAlgo::PSATD) {
+    } else if (electrostatic_solver_id  != ElectrostaticSolverAlgo::None ||
+               electromagnetic_solver_id == ElectromagneticSolverAlgo::PSATD) {
         // Computation of dt for spectral algorithm
         // (determined by the minimum cell size in all directions)
-#if defined(WARPX_DIM_1D_Z)
-        deltat = cfl * dx[0] / PhysConst::c;
-#elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
-        deltat = cfl * std::min(dx[0], dx[1]) / PhysConst::c;
-#else
-        deltat = cfl * std::min(dx[0], std::min(dx[1], dx[2])) / PhysConst::c;
-#endif
+        if (m_max_dt.has_value() && electromagnetic_solver_id == ElectromagneticSolverAlgo::None) {
+            deltat = m_max_dt.value();
+        } else {
+            deltat = cfl / PhysConst::c * minDim(dx);
+        }
```

Comment:
I think it would be more readable to split this into two `if`-blocks, i.e., something like:
```
} else if (electrostatic_solver_id != ElectrostaticSolverAlgo::None) {
    // Set dt for electrostatic algorithm
    if (m_max_dt.has_value()) {
        deltat = m_max_dt.value();
    } else {
        deltat = cfl / PhysConst::c * minDim(dx);
    }
} else if (electromagnetic_solver_id == ElectromagneticSolverAlgo::PSATD) {
    // Computation of dt for spectral algorithm
    deltat = cfl / PhysConst::c * minDim(dx);
}
```
It does repeat one line of code but I think the logic is much easier to read.